proposed/building-arch-specific-applications.md

```diff
@@ -0,0 +1,806 @@
+* Defaults should be the same between `build` and `publish`.
+* All common scenarios should work without specifying a RID if you are happy targeting the current machine.
+* We should strongly avoid any breaking changes.
+
+### Proposed Application Types
+
+For the purposes of this document, there are three types of applications:
+
+* **Architecture-neutral, framework-dependent** -- Applications of this kind target and can run on multiple architectures. They require an installed runtime to run (that matches the architecture of the target machine). They carry multiple sets of dependent native libraries for a variety of target architectures. All application and other architecture-agnostic  binaries are contained in a single directory and architecture-specific binaries are carried in RID-specific directories within a `runtimes` directory.
+* **Architecture-specific, framework-dependent** -- Applications of this kind target and can only run on a single architecture, like Windows x64 or Linux ARM32. They require an installed runtime to run. They carry one set of dependent native libraries for a single target architecture. They include an executable launcher for the given architecture. All application binaries are contained in a single directory.
+* **Architecture-specific, self-contained** -- Applications of this kind target and can only run on a single architecture, like Windows x64 or Linux ARM32. They carry a runtime and one set of dependent native libraries for a single target architecture. They include an executable launcher for the given architecture. All application, runtime and native binaries are contained in a single directory.
+
+### Proposed CLI argument and MSBuild property changes
+
+The following changes are proposed for .NET Core 3.0. `build` will be the specific verb discussed in the following text. The behavior is intended to be the same for `publish` unless specifically called out.
+
+#### Framework-dependent architecture-netural applications
+
+Maintain framework-dependent architecture-netural applications as the default for `build`.
+  
+* Produces an application with native dependencies for all supported architectures.
+* Remove the capability to produce EXEs for this type of build.
+* If EXEs are desired for this build type, consider for a later release. It's a hard scenario to get right due to filename conflicts.
+
+ASP.NET and console templates should accept this default.
+
+#### Framework-dependent architecture-specific applications
+
+Expose a new opt-in build type, called architecture-specific framework-dependent.
+
+* Produces an application with native dependencies and an EXE for a single specified architecture.
+* This build type can be selected via
+  * MSBuild: `<ArchitectureSpecific>true</ArchitectureSpecific>`
+  * CLI: `--architecture-specific` or `-a`
+* A RID must be specified (or it is an error). Any of the following options is legal to specify a RID:
```

Comment:
Why wouldn't we use the current OS to find a default RID?

Reply:
It means that the build will produce different results on different machines, and some configurations will produce the wrong results. Some people will build Windows Forms applications on a Linux x64 machine, but will still want Windows x64 assets. "Current Arch" should be opt-in. It is documented in the spec.